dbms/src/Functions/tests/gtest_weekofyear.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/MyTime.h>
+#include <Functions/FunctionFactory.h>
+#include <TestUtils/FunctionTestUtils.h>
+#include <TestUtils/TiFlashTestBasic.h>
+
+#include <string>
+#include <vector>
+
+namespace DB::tests
+{
+class TestWeekOfYear : public DB::tests::FunctionTest
+{
+};
+
+TEST_F(TestWeekOfYear, WeekOfYear)
+try
+{
+    DAGContext * dag_context = context.getDAGContext();
+    UInt64 ori_flags = dag_context->getFlags();
```

Comment:
Maybe move the flag related operations to TestWeekOfYear's setup and  Teardown methods.

Reply:
Emmm, there is only one test and the flag is specifically used for testing zero date.